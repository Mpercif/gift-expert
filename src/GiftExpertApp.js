import React, { useState } from 'react'
import AddCategory from './Components/AddCategory'
import GiftGrid from './Components/GiftGrid'


const GiftExpertApp = ({defaultCategories = []}) => {

    

    const [categorias, setCategorias] = useState(defaultCategories)

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategory = {setCategorias} />
            <hr/>
            <ol>
                {categorias.map(cat=> 
                    <GiftGrid
                        key={cat} 
                        category={cat}  
                    />)
                }
            </ol>
        </>
    )
}


export default GiftExpertApp
