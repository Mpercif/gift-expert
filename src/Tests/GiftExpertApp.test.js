import { shallow } from "enzyme"
import '@testing-library/jest-dom'
import React from 'react';

import GiftExpertApp from "../GiftExpertApp"

describe('Pruebas al componenten GiftExpertApp', () => {

    test('Rendenrizacion del componente ', () => {
        
        const wrapper = shallow(<GiftExpertApp /> )
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar una lista de categoria ', () => {
        
        const categories = ['One Punch', 'Boku No Hero']

        const wraper = shallow(<GiftExpertApp  defaultCategories={categories} />)

        expect(wraper).toMatchSnapshot()
        expect(wraper.find('GiftGrid').length).toBe(categories.length)
    })
    

    
    
})
