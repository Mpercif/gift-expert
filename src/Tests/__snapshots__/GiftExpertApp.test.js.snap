// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componenten GiftExpertApp Debe de mostrar una lista de categoria  1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="One Punch"
      key="One Punch"
    />
    <GiftGrid
      category="Boku No Hero"
      key="Boku No Hero"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al componenten GiftExpertApp Rendenrizacion del componente  1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
